From 4e903e7b4e80577164c84e92e7d12b5f8f314132 Mon Sep 17 00:00:00 2001
From: Jan Stancek <jstancek@redhat.com>
Date: Wed, 15 Feb 2017 14:10:18 +0100
Subject: [PATCH 3/5] ld.hugetlbfs: pick an emulation if -m is not present

If -m is not passed on command line $EMU ends up empty
and as result HPAGE_SIZE and SLICE_SIZE are left uninitialized.

Try environment variable LDEMULATION and if it's not defined
pick first from "Supported emulations" list of ld -V.

Signed-off-by: Jan Stancek <jstancek@redhat.com>
Signed-off-by: Eric B Munson <emunson@mgebm.net>
---
 ld.hugetlbfs | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/ld.hugetlbfs b/ld.hugetlbfs
index 2dda451..8ee917b 100755
--- a/ld.hugetlbfs
+++ b/ld.hugetlbfs
@@ -92,6 +92,17 @@ if [ -z "$EMU" ]; then
 	fi
 fi
 
+# if -m is not present on command line
+if [ -z "$EMU" ]; then
+	if [ -n "$LDEMULATION" ]; then
+		# try env. variable
+		EMU="$LDEMULATION"
+	else
+		# pick first supported
+		EMU="$(ld -V | sed -n '/Supported emulations/{n;p}' | tr -d ' ')"
+	fi
+fi
+
 MB=$((1024*1024))
 case "$EMU" in
 elf32ppclinux|elf64ppc)	HPAGE_SIZE=$((16*$MB)) SLICE_SIZE=$((256*$MB)) ;;
-- 
1.8.3.1

